git-hub

```diff
@@ -87,15 +103,292 @@ def push():
+        if(pr_dictionary is None):
+            open_PR = "C"
+            key, pr_dictionary = find_in_dictionary(pr_num, closed_dict)
+        if(pr_dictionary is None):
+            click.echo("Could not find PR #{}. Run 'git hub sync' and try again.".format(pr_num))
+        else:
+            reviewers = "None"
+            if(pr_dictionary['reviewers'] is not ""):
+                reviewers = pr_dictionary['reviewers']
+            assignee = "None"
+            if(pr_dictionary['assignee'] is not ""):
+                assignee = pr_dictionary['assignee']
+            milestone = "None"
+            if(pr_dictionary['milestone'] is not ""):
+                milestone = pr_dictionary['milestone']
+            labels = "None"
+            if(pr_dictionary['labels'] is not ""):
+                labels = pr_dictionary['labels']
+            click.echo('{0} \x1b[0;32;40m {1} \x1b[0m {2}/{3} {4}'.format(key, open_PR, pr_dictionary['user'], pr_dictionary['branch'], pr_dictionary['comment']))
+            click.echo('-Labels: {0}  -Reviewers: {1}  -Assignees: {2}  -Milestones: {3}'.format(labels, reviewers, assignee, milestone))
+    except (OSError, IOError) as e:
+        # if pull-requests.toml hasnt been created yet calls sync and then reties to fetch
+        sync()
+        get_info(pr_num)
+
+
+def print_in_order(sort, q):
+    """
+    Prints items in q sorted by order given by sort
+
+    Parameters
+    ----------
+    sort : string
+        sort in increasing or decreasing order
+    q : PriorityQueue
+        items to be sorted, currently in increasing order
+    """
+    if(sort.startswith('i')):
+        while not q.empty():
+            pair = q.get()
+            click.echo("{0} : {1}".format(pair[1], pair[0][:10]))
+    else:
+        stack = []
+        while not q.empty():
+            pair = q.get()
+            stack.append("{0} : {1}".format(pair[1], pair[0][:10]))
+        while not len(stack) == 0:
+            click.echo(stack.pop())
+
+
+def findMatch(open_or_closed, list_of_dictionaries, keyword, user, comment, number, branch, label, appeared_before, line):
+    """Helper function that finds all matches in the given dictionary that fits the specified parameters
+
+    Parameters
+    ----------
+    open_or_closed : boolean
+        Whether looking in closed PR or open PR
+    dictionary : dictionary
+        pull-requests.toml information mapping PR number to info
+    keyword : string
+        Searches if any part of user, branch, comment, or number match
+    user : string
+        Search by PR username
+    comment : string
+        Search by PR comment
+    number : string
+        Search by PR number
+    branch : string
+        Search by PR branch
+    appeared_before : boolean
+        Whether the search match is the first match
+    line : boolean
+        Whether to draw lines before listing closed pull requests
+    """
+    q = Q.PriorityQueue()
```

Comment:
I think we can replace the content of this function with a generic dictionary matching function.  Something like:

`filter_dictionary(PRs, constraints)`

where `PRs` is a list of PR entries, and `constraints` is a dictionary.